src/scripts/update-price-oracle.js

```diff
@@ -0,0 +1,218 @@
+    }) => {
+      console.log('  - %s (%s) from %s to %s', symbol, address, priceContract, price)
+    })
+
+    const { tokens, numerators, denominators } = tokenBatch.reduce((acc, { address, price }) => {
+      acc.tokens.push(address)
+      acc.numerators.push(price.numerator)
+      acc.denominators.push(price.denominator)
+
+      return acc
+    }, {
+      tokens: [],
+      numerators: [],
+      denominators: []
+    })
+
+
+    // console.log('tokens', tokens)
+    // console.log('numerators', numerators)
+    // console.log('denominators', denominators)
+    if (dryRun) {
+      await fixedPriceOracle.setPrices.call(tokens, numerators, denominators)
+      console.log('Dry run: %d token prices call succeed', tokenBatch.length)
+    } else {
+      const setPriceResult = await fixedPriceOracle.setPrices(tokens, numerators, denominators)
+      console.log('%d token prices were updated. Transaction: %s', tokenBatch.length, setPriceResult.tx)
+    }
+  }
+
+  console.log('\n **************  All prices were updated  **************\n')
+}
+
+async function getTokensToUpdate ({
+  fixedPriceOracle,
+  tokens
+}) {
+  // Get the actual prices for the tokens in the smart contract
+  const actualPricePromises = tokens.map(async token => {
+    const { price: priceDecimal, ...tokenFields } = token
+
+    // Get price in the contract
+    const priceContract = await getPriceValue({
+      address: token.address,
+      fixedPriceOracle
+    })
+
+    // Get price
+    const price = new Fraction(priceDecimal)
+
+    return {
+      ...tokenFields,
+      priceDecimal,
+      price,
+      priceContract
+    }
+  })
+
+  // Wait for all the prices of the tokens
+  const actualPrices = await Promise.all(actualPricePromises)
+
+  const { tokensAlreadyUpdated, tokensToUpdate } = actualPrices.reduce((acc, token) => {
+    // console.log('%s: %s = %s? %s', token.symbol, token.price, token.priceContract, token.price.equals(token.priceContract))
+    if (token.price.equals(token.priceContract)) {
+      acc.tokensAlreadyUpdated.push(token)
+    } else {
+      acc.tokensToUpdate.push(token)
+    }
+    return acc
+  }, {
+    tokensAlreadyUpdated: [],
+    tokensToUpdate: []
+  })
+
+  if (tokensAlreadyUpdated.length > 0) {
+    console.log(
+      '%d tokens are already updated:\n  %s',
+      tokensAlreadyUpdated.length,
+      tokensAlreadyUpdated.map(token => token.symbol).join(', ')
+    )
+  } else {
+    console.log('All tokens need to be updated')
+  }
+
+  if (tokensToUpdate.length > 0) {
+    console.log(
+      '\n%d tokens need to be updated:\n  %s',
+      tokensToUpdate.length,
+      tokensToUpdate.map(token => token.symbol).join(', ')
+    )
+  } else {
+    console.log("\nThere's nothing to do")
+  }
+
+  return tokensToUpdate
+}
+
+async function getPriceValue({
+  address,
+  fixedPriceOracle
+}) {
+  const { 
+    0: numeratorBn,
+    1: denominatorBn 
+  } = await fixedPriceOracle.getPriceValue(address) // { 0: 389, 1: 2500 }
```

Comment:
Not `const [numeratorBn, denominatorBn = ...`?
Maybe even `let [num, den] = (await ...).map(bn => bn.toNumber())`

Reply:
same issue as above, got you

Reply:
I thought about that, and it's good, but I usually try to avoid using await in parenthesis, in my opinion it can be sometimes a bit confusing and error prone. If you forget the parentesis, u will await an undefined, because is the value for the property of an unresolved promise. This in some cases will continue without breaking

For this case it could be done as u suggest, is not big deal